services/thingsearch/persistence/src/main/java/org/eclipse/ditto/services/thingsearch/persistence/write/streaming/EnforcementFlow.java

```diff
@@ -185,8 +222,14 @@ public static EnforcementFlow of(final StreamConfig updaterStreamConfig,
             return getEnforcer(metadata, thing)
                     .map(entry -> {
                         if (entry.exists()) {
-                            return EnforcedThingMapper.toWriteModel(thing, entry.getValueOrThrow(), entry.getRevision(),
-                                    maxArraySize);
+                            try {
+                                return EnforcedThingMapper.toWriteModel(thing, entry.getValueOrThrow(),
+                                        entry.getRevision(),
+                                        maxArraySize);
+                            } catch (JsonRuntimeException e) {
+                                log.error(e.getMessage(), e);
+                                return ThingDeleteModel.of(metadata);
+                            }
```

Comment:
Why do we need to catch a JsonRuntimeException here?

Reply:
If a property that is causing an exception is passed, the searchUpdater suspends. Catching JsonRuntimeExceptions allows the searchUpdater to keep working, while just deleting the thing, corresponding to the exception, from search. I would suggest keeping this to ensure searchUpdater keeps working as it should in case of an exception.

Reply:
Please make exceptions in catch clauses "final" (and generally everything else which is not required to change).